### DISCUSSION TITLE
How to generate the captcha token with the CLI?

### DISCUSSION THREAD

**Discussion by HeraldCoupe** [bimmerconnected/bimmer_connected#676] [Q&A]
Sorry if I'm being dim but I need some help with the new CLI please.
I've installed v0.17 and I've read the documentation but I can't figure out how to generate the captcha token.
Please can you give me some advice on the syntax of the command to do this?

**Comment by rikroe** [comment]
I thought its quite clear with the big blue box in https://bimmer-connected.readthedocs.io/en/stable/cli.html, forwarding to https://bimmer-connected.readthedocs.io/en/stable/captcha.html and then you have to select your region.

If you have suggestions to make this easier to understand, please let me know.

**Reply by HeraldCoupe** [reply]
Thanks for your reply. The problem I'm having is with the syntax for the 'first login'. I tried this:
![bimmerconnected error](https://github.com/user-attachments/assets/71e086f7-6816-4797-a6d9-21af4c625ee6)
which obviously error'd.
I also tried this:
![bimmerconnected error01](https://github.com/user-attachments/assets/a4bf45d3-e289-4b0f-9147-26a3feaf7b88)
which also resulted in an error.

My suggestion is to include an example of the syntax needed for the 'first login'. 
For example, something like this:

1. First login: bimmerconnected --debug --captcha-token status _username password_ rest_of_world

Subsequent examples of the syntax needed would also be useful but I've not got that far yet.

I appreciate you taking the time to look into this for me.

**Reply by rikroe** [reply]
Something like this? 
https://bimmer-connected.readthedocs.io/en/stable/captcha.html#using-the-cli

**Reply by HeraldCoupe** [reply]
I'm sorry but I've obviously not explained the problem well enough. I'll start again.

I've updated Bimmerconnected to v0.17. 

Please correct me if I'm wrong or have missed something but at the top of https://bimmer-connected.readthedocs.io/en/stable/captcha.html#using-the-cli it states that 'The first login requires a captcha to be solved.' but there is no example of how to run a 'first login' to generate the token.

Any help is much appreciated.

**Reply by rikroe** [reply]
Ah, I see where you are stuck. I'll try to think of a way to make it clearer.

For now, do the 3 red steps in the screenshot below help you?

![Screenshot 2024-11-23 154131](https://github.com/user-attachments/assets/7a951edb-b877-49e2-a32e-f8d8caa98892)

**Reply by HeraldCoupe** [reply]
Thank you so much. I feel a bit foolish because I thought the screen to generate the captcha was a screenshot. I didn't know I needed to click in the box to generate the token.

I then struggled with step 2 but I'm pretty sure that's because I wasn't quick enough (I think the token expires quickly) But once I figured it out it worked and subsequent logins (step 3) are working perfectly.

Thanks again for your help and hard work in getting Bimmer Connected working again.

**Reply by rikroe** [reply]
Great if it works now!

Would replacing the submit button with "validate captcha and generate token" help?

**Reply by HeraldCoupe** [reply]
I think that all you need to do is include the line "Click in the 'I am human' checkbox and press Submit to generate the captcha token".
